app/components/work_package_relations_tab/relations_mediator.rb

```diff
@@ -29,9 +29,21 @@
 #++
 
 class WorkPackageRelationsTab::RelationsMediator
-  RelationGroup = Data.define(:type, :visible_relations, :ghost_relations) do
-    def initialize(type:, visible_relations:, ghost_relations:)
-      super(type: ActiveSupport::StringInquirer.new(type), visible_relations:, ghost_relations:)
+  RelationGroup = Data.define(:type, :work_package, :visible_relations, :ghost_relations, :closest_relation) do
+    def initialize(type:, work_package:, visible_relations:, ghost_relations:)
+      type = ActiveSupport::StringInquirer.new(type.to_s)
+      closest_relation = self.class.closest_relation(type, visible_relations + ghost_relations)
+      super(type:, work_package:, visible_relations:, ghost_relations:, closest_relation:)
+    end
+
+    def self.closest_relation(type, relations)
+      return nil unless type.follows?
+
```

Comment:
```suggestion
   def self.closest_relation(relations)
```

We could also make this a `private_class_method`.

Reply:
That said, it might be even more sense to move the closest logic to `ClosestRelation`, e.g.

```ruby
    def self.of(relations)
      relations
        .map { new(it) }
        .select(&:soonest_start)
        .max
        &.relation
    end
```

Reply:
I like the idea of moving it to `ClosestRelation`. I'll try that.